tests/kernel/context/src/main.c

```diff
@@ -26,7 +26,7 @@
 #include <zephyr/irq_offload.h>
 #include <zephyr/sys_clock.h>
 
-#if defined(CONFIG_SOC_POSIX) || defined(CONFIG_ARC)
+#if defined(CONFIG_SOC_POSIX)
 /* TIMER_TICK_IRQ <soc.h> header for certain platforms */
 #include <soc.h>
```

Comment:
so for posix soc.h is still allowed, but for ARC not ;-)

Reply:
It's not about ARC, please read the PR for more context. ARM64 has already been cleanup, and others will gradually come. I've started with small platforms, others like ARM will take more time.

Reply:
Drive-by comment:  this test is just awful, literally having to punch a hole through into every timer driver in the whole tree, which is why the soc.h madness exists here in a platform-dependent way.

I had a patch sitting around as part of the MTL audio DSP work that just defined a global variable to hold the IRQ (which would then be set by the drivers, ideally in a static initializer such that it optimizes out if unused), which eliminated all the complexity.  But I never got around to making it work for all the drivers.

Reply:
> Drive-by comment:  this test is just awful, literally having to punch a hole through into every timer driver in the whole tree, which is why the soc.h madness exists here in a platform-dependent way.
> 
> 
> 
> I had a patch sitting around as part of the MTL audio DSP work that just defined a global variable to hold the IRQ (which would then be set by the drivers, ideally in a static initializer such that it optimizes out if unused), which eliminated all the complexity.  But I never got around to making it work for all the drivers.

It would definitely be nice to clean this up.